plugins/kubernetes/app/models/kubernetes/role.rb

```diff
@@ -60,6 +60,20 @@ def self.configured_for_project(project, git_sha)
       end
     end
 
+    def defaults
+      return unless raw_template = project.repository.file_content(config_file, 'HEAD', update: false)
+      objects = Array.wrap(Kubernetes::Util.parse_file(raw_template, config_file))
+      return unless deploy = objects.detect { |o| ['Deployment', 'DaemonSet'].include?(o.fetch('kind')) }
```

Comment:
I think this logic already exists in ReleaseDoc or DeployYAML or something.  DRY if possible?